vm_core.h

```diff
@@ -256,9 +256,19 @@ STATIC_ASSERT(sizeof_iseq_inline_constant_cache_entry,
 
 struct iseq_inline_constant_cache {
     struct iseq_inline_constant_cache_entry *entry;
-    // For YJIT: the index to the opt_getinlinecache instruction in the same iseq.
-    // It's set during compile time and constant once set.
-    unsigned get_insn_idx;
+
+    /**
+     * A null-terminated list of ids, used to represent a constant's path
+     * idNULL is used to represent the :: prefix, and 0 is used to donate the end
+     * of the list.
+     *
+     * For example
+     *   FOO        {rb_intern("FOO"), 0}
+     *   FOO::BAR   {rb_intern("FOO"), rb_intern("BAR"), 0}
+     *   ::FOO      {idNULL, rb_intern("FOO"), 0}
+     *   ::FOO::BAR {idNULL, rb_intern("FOO"), rb_intern("BAR"), 0}
+     */
+    ID *segments;
```

Comment:
I think this should be constified, and related functions.
```suggestion
    const ID *segments;
```

Reply:
👍 done